sklearn/feature_selection/tests/test_rfe.py

```diff
@@ -141,6 +144,31 @@ def test_rfe():
     assert_array_almost_equal(X_r, X_r_sparse.toarray())
 
 
+def test_rfe_regression():
+    X, y, coef = make_regression(n_samples=100, n_features=10,
+                                 n_informative=2, coef=True, random_state=42)
+    X_sparse = sparse.csr_matrix(X)
+    rfe = RFE(estimator=LinearRegression(), n_features_to_select=2,
+              step=0.1)
+    rfe.fit(X, y)
+    X_r = rfe.transform(X)
+
+    rfe_sparse = RFE(estimator=LinearRegression(), n_features_to_select=2,
+                     step=0.1)
+    rfe_sparse.fit(X_sparse, y)
+    X_r_sparse = rfe_sparse.transform(X_sparse)
+
+    assert_equal(len(rfe.ranking_), X.shape[1])
+    assert_equal(X_r.shape[1], 2)
+
+    clf = LinearRegression()
+    clf.fit(X_r, y)
+    assert_array_almost_equal(X_r, X[:, coef > 0])
```

Comment:
`coef > 0 ->  coef !=`

as it makes sense to have negative coefs in regression.